app/assets/javascripts/discourse/app/components/discovery-topics-list.js

```diff
@@ -27,12 +27,16 @@ export default Component.extend(UrlRefresh, LoadMore, {
 
   @on("didInsertElement")
   _monitorTrackingState() {
-    this.topicTrackingState.onStateChange(() => this._updateTrackingTopics());
+    this.stateChangeCallbackId = this.topicTrackingState.onStateChange(() =>
+      this._updateTrackingTopics()
+    );
   },
 
   @on("willDestroyElement")
   _removeTrackingStateChangeMonitor() {
-    this.topicTrackingState.offStateChange(this.stateChangeCallbackId);
+    if (this.stateChangeCallbackId) {
+      this.topicTrackingState.offStateChange(this.stateChangeCallbackId);
+    }
```

Comment:
Couldn't `offStateChange` handle directly the id check?

This is something Ember does for `cancel` for example and I think it makes sense.